include/libvfio-user.h

```diff
@@ -436,24 +436,10 @@ void
 vfu_destroy_ctx(vfu_ctx_t *vfu_ctx);
 
 /**
- * Once the vfu_ctx is configured vfu_ctx_drive() drives it. This function waits
- * for commands coming from the client, and processes them in a loop.
- *
- * @vfu_ctx: the libvfio-user context to drive
- *
- * @returns 0 on success, -errno on failure.
- */
-int
-vfu_ctx_drive(vfu_ctx_t *vfu_ctx);
-
-/**
- * Polls, without blocking, an vfu_ctx. This is an alternative to using
- * a thread and making a blocking call to vfu_ctx_drive(). Instead, the
- * application can periodically poll the context directly from one of
- * its own threads.
- *
- * This is only allowed when LIBVFIO_USER_FLAG_ATTACH_NB is specified during
- * creation.
+ * Polls the vfu_ctx and processes the command recieved from client.
+ * Non-blocking poll if vfu_ctx is created with LIBVFIO_USER_FLAG_ATTACH_NB,
+ * otherwise blocking. With non-blocking poll application can periodically poll
+ * the context directly from one of its own threads.
```

Comment:
I just realized that the name of the function can be misleading: in blocking mode the function keeps processing requests until failure or the client closes the connection. In this case the semantics of the function are very different from the standard select(2) and poll(2), which tell you that some file descriptor is reading and it's up to the caller to consume the file descriptor data. I think a more suitable name would be `process_request` or something similar. I should have realized that earlier.

Reply:
Agreed this needs to be renamed suitably. I am fine with vfu_process_request(). @jlevon any better suggestions?

Reply:
"process_request" implies it does only one request.
How about vfu_ctx_run() ?

Reply:
Yes that also sounds good. Maybe we can call it as vfu_run_ctx(), to be in sync with vfu_*_ctx.

Reply:
I missed the likes, no notifications for emoji's :)
Renamed vfu_ctx_poll() to vfu_run_ctx(), @tmakatos have another look